src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -86,8 +63,35 @@
   If the query can't be parsed, a 400 is returned."
   [entity version query paging-options db url-prefix]
   (try
-    (let [parsed-query (json/parse-strict-string query true)]
-      (stream-query-result entity version parsed-query paging-options db url-prefix http/stream-json-response))
+    (let [parsed-query (json/parse-strict-string query true)
+          query-error (promise)
+          count-query (promise)
```

Comment:
another thing worth considering is that row counts are now available directly through the query language. I'm not exactly sure what people are using --include-total for, but if just calling the count function would suit their needs we could consider deprecating it.